cd/utils/artifact_repository.py

```diff
@@ -0,0 +1,615 @@
+        logger.debug('Could not determine git commit id')
+    else:
+        logger.debug('Commit id is: {}'.format(commit_id))
+    return commit_id
+
+
+def get_linux_os_release_properties() -> Optional[Dict[str, str]]:
+    """
+    Makes a dictionary out of /etc/os-release
+    :return: A dictionary of os release properties
+    """
+    logger.debug('Extracting operating system properties from /etc/os-release')
+    if not os.path.isfile('/etc/os-release'):
+        logger.debug('Error: /etc/os-release not found')
+        return None
+
+    try:
+        with open('/etc/os-release', 'r') as fp:
+            # removes empty spaces and quotation marks from line
+            property_tuple_list = [line.strip().replace('"', '').split('=') for line in fp if line.strip()]
+            return {key: value for (key, value) in property_tuple_list}
+    except Exception as e:
+        logger.debug('Error parsing /etc/os-release')
+        logger.debug(e)
+        return None
+
+
+def get_linux_distribution_and_version() -> Optional[str]:
+    """
+    Returns the linux distribution and version by taking
+    the values of ID and VERSION_ID from /etc/os-release and
+    concatenating them. Eg. centos7, ubuntu16.04, etc.
+    :return: The linux distribution and version string, or None if not found.
+    """
+    logger.debug('Getting linux distribution and version')
+    os_properties = get_linux_os_release_properties()
+    if os_properties:
+        logger.debug('os properties: {}'.format(os_properties))
+        distribution = os_properties['ID']
+        version = os_properties['VERSION_ID']
+        return "{}{}".format(distribution, version)
+
+    logger.debug('Error getting linux distribution and version. Could not determine os properties.')
+    return None
+
+
+def probe_operating_system() -> str:
+    """
+    Probes the system to determine the operating system.
+    :return: The name of the operating system, e.g. win32, darwin, ubuntu16.04, centos7, etc.
+    """
+    logger.debug('Determining operating system')
+    operating_system = sys.platform
+    logger.debug('Found platform: {}'.format(operating_system))
+    if operating_system.startswith('linux'):
+        operating_system = get_linux_distribution_and_version()
+
+    logger.debug('Operating system is {}'.format(operating_system))
+    return operating_system
+
+
+def get_libmxnet_features(libmxnet_path: str) -> Optional[Dict[str, bool]]:
+    """
+    Returns a string -> boolean dictionary mapping feature name
+    to whether it is enabled or not
+    :param libmxnet_path: path to the libmxnet library
+    :return: dictionary of features to whether they are enabled
+    """
+    logger.debug('Getting feature dictionary from {}'.format(libmxnet_path))
+
+    class Feature(ctypes.Structure):
+        _fields_ = [("_name", ctypes.c_char_p), ("enabled", ctypes.c_bool)]
+
+        @property
+        def name(self):
+            return self._name.decode()
+
+    try:
```

Comment:
Are you trying to avoid loading mxnet since you are not using directly the python binding?  Maybe add a comment why loading MXNet is not desirable.

Reply:
Added =D